In [1]:
import pandas as pd
import numpy as np

# Simulando dados
np.random.seed(42)
n_usuarios = 1000

# Grupo Controle (Layout Antigo) - Conversão de 10%
controle = pd.DataFrame({
    'grupo': 'controle',
    'conversao': np.random.binomial(1, 0.10, n_usuarios)
})

# Grupo Tratamento (Layout Novo) - Conversão de 13%
tratamento = pd.DataFrame({
    'grupo': 'tratamento',
    'conversao': np.random.binomial(1, 0.13, n_usuarios)
})

df = pd.concat([controle, tratamento])

In [2]:
from statsmodels.stats.proportion import proportions_ztest

# Contando sucessos e total de observações
conversoes_controle = df[df['grupo'] == 'controle']['conversao'].sum()
conversoes_tratamento = df[df['grupo'] == 'tratamento']['conversao'].sum()

n_controle = df[df['grupo'] == 'controle'].shape[0]
n_tratamento = df[df['grupo'] == 'tratamento'].shape[0]

# Executando o teste estatístico
sucessos = [conversoes_controle, conversoes_tratamento]
observacoes = [n_controle, n_tratamento]

z_stat, p_valor = proportions_ztest(sucessos, observacoes)

print(f'Estatística Z: {z_stat:.2f}')
print(f'P-Valor: {p_valor:.4f}')

Estatística Z: -2.17
P-Valor: 0.0301


## 📊 Conclusão do Teste

Após realizar o Teste A/B com 2.000 usuários (1.000 em cada grupo), obtivemos os seguintes resultados:   

**- Estatística Z:** -2.17  
**- P-Valor:** 0.0301  

**Interpretação Estatística:**  
Dado que o p-valor (0.0301) ficou abaixo do limite crítico de 0.05, rejeitamos a hipótese nula. 
Isso indica que a alteração no layout do botão de checkout causou uma mudança estatisticamente significante na taxa de conversão.

**Recomendação de Negócio:**  
Com 95% de confiança, recomendamos a implementação definitiva do novo layout (Grupo de Tratamento), pois ele apresentou uma performance superior ao modelo atual. 
Estima-se que essa mudança trará um impacto positivo recorrente nas vendas da plataforma.
